node/tests/SharedTests.ts

```diff
@@ -2367,6 +2367,53 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.only.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "renamenx test_%p",
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key1 = `{key}-1-${uuidv4()}`;
+                const key2 = `{key}-2-${uuidv4()}`;
+                const key3 = `{key}-3-${uuidv4()}`;                                
+
+                // renamenx missing key
+                try {
+                    expect(await client.renameNX(key1, key2)).toThrow();
+                } catch (e) {
+                    expect((e as Error).message).toMatch(
+                        "no such key",
+                    );
+                }
+
+                // renamenx a string
+                await client.set(key1, "key1");
+                await client.set(key3, "key3");                                
+                expect(await client.renameNX(key1, key2)).toEqual(true);
+                expect(await client.renameNX(key2, key3)).toEqual(false);
+                const result2 = await client.get(key2);
+                expect(result2).toEqual("key1");
+                await client.del([key1, key2]);
+
+                // this one remains unchanged
+                const result3 = await client.get(key3);
+                expect(result3).toEqual("key3");
+
+                // Same-slot requirement
```

Comment:
Add this command to existing test in https://github.com/aws/glide-for-redis/blob/64d2273f093a6f27d0674f741f6ea7303b02563b/node/tests/RedisClusterClient.test.ts#L279